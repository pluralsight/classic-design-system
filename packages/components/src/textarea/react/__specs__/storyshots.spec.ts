import path from 'path'
import initStoryshots, {
  snapshotWithOptions
} from '@storybook/addon-storyshots'

const createNodeMock = () => document.createElement('div')

initStoryshots({
  configPath: path.resolve(__dirname, '../../.storyshots'),
  framework: 'react',
  test: snapshotWithOptions({ createNodeMock })
})
