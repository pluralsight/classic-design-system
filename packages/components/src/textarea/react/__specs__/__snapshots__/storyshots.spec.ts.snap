// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Components/TextArea All Labels 1`] = `
Array [
  <label
    data-css-na8jqt=""
  >
    <div
      data-css-9kywkh=""
    >
      Some label
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          placeholder="Some placeholder"
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Some sub label
    </div>
  </label>,
  <label
    data-css-na8jqt=""
  >
    <div
      data-css-9kywkh=""
    >
      Some label
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-8khcid=""
          disabled={false}
          placeholder="Some placeholder"
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Some sub label
    </div>
  </label>,
]
`;

exports[`Storyshots Components/TextArea Disabled Compare 1`] = `
Array [
  <label
    data-css-na8jqt=""
  >
    <div
      data-css-9kywkh=""
    >
      Normal
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          placeholder="I'm normal, see"
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Still normal
    </div>
  </label>,
  <label
    data-css-1tctpgo=""
  >
    <div
      data-css-9kywkh=""
    >
      I'm not usable
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={true}
          placeholder="I'm untouchable"
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Neither am I
    </div>
  </label>,
]
`;

exports[`Storyshots Components/TextArea Error Compare 1`] = `
Array [
  <label
    data-css-na8jqt=""
  >
    <div
      data-css-9kywkh=""
    >
      Normal
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          placeholder="I'm normal, see"
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Still normal
    </div>
  </label>,
  <label
    data-css-na8jqt=""
  >
    <div
      data-css-9kywkh=""
    >
      I'm in an error state
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-in6gqi=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          style={
            Object {
              "height": 112,
            }
          }
          value="Wow, here is all the content wrapped up inside.Wow, here is all the content wrapped up inside.Wow, here is all the content wrapped up inside.   "
        />
      </div>
      <div
        data-css-h6j4b2=""
      >
        <div
          data-css-7dab2f=""
        >
          <svg
            aria-label="warning icon"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 2c-4.411 0-8 3.589-8 8s3.589 8 8 8 8-3.589 8-8-3.589-8-8-8zm.5 14h-1a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v1a.5.5 0 01-.5.5zm0-4h-1a.5.5 0 01-.5-.5V7a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v6.5a.5.5 0 01-.5.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      data-css-tayj8l=""
    >
      Here
    </div>
  </label>,
]
`;

exports[`Storyshots Components/TextArea Focus 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Full Width Layout 1`] = `
<div
  style={
    Object {
      "border": "1px solid blue",
      "width": "800px",
    }
  }
>
  <label
    data-css-na8jqt=""
    style={
      Object {
        "display": "block",
        "width": "100%",
      }
    }
  >
    <div
      data-css-9kywkh=""
    >
      max-width still applies
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-ctre2o=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
    </div>
  </label>
  <label
    data-css-na8jqt=""
    style={
      Object {
        "display": "block",
        "maxWidth": "none",
        "width": "100%",
      }
    }
  >
    <div
      data-css-9kywkh=""
    >
      remove max-width
    </div>
    <div
      data-css-1ios7mb=""
    >
      <div
        data-css-in6gqi=""
      >
        <textarea
          data-css-1paz7u7=""
          disabled={false}
          style={
            Object {
              "height": 112,
            }
          }
        />
      </div>
      <div
        data-css-h6j4b2=""
      >
        <div
          data-css-7dab2f=""
        >
          <svg
            aria-label="warning icon"
            role="img"
            viewBox="0 0 24 24"
          >
            <path
              clipRule="evenodd"
              d="M12 2C6.486 2 2 6.486 2 12s4.486 10 10 10 10-4.486 10-10S17.514 2 12 2zm0 2c-4.411 0-8 3.589-8 8s3.589 8 8 8 8-3.589 8-8-3.589-8-8-8zm.5 14h-1a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v1a.5.5 0 01-.5.5zm0-4h-1a.5.5 0 01-.5-.5V7a.5.5 0 01.5-.5h1a.5.5 0 01.5.5v6.5a.5.5 0 01-.5.5z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </label>
</div>
`;

exports[`Storyshots Components/TextArea Label 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-9kywkh=""
  >
    Some label
  </div>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Label And Sub Label 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-9kywkh=""
  >
    Some label
  </div>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
  <div
    data-css-tayj8l=""
  >
    Some sublabel
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Less Than Four Rows 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-9kywkh=""
  >
    Min height engages
  </div>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 40,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea More Than Four Rows 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-9kywkh=""
  >
    6 rows of text
  </div>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 160,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea No Label 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Placeholder 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        placeholder="some placeholder"
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Prop Name 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        name="myFieldNameOfPower"
        placeholder="I have a form name"
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Prop On Change 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        placeholder="Focus on me!"
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
</label>
`;

exports[`Storyshots Components/TextArea Sub Label 1`] = `
<label
  data-css-na8jqt=""
>
  <div
    data-css-1ios7mb=""
  >
    <div
      data-css-ctre2o=""
    >
      <textarea
        data-css-1paz7u7=""
        disabled={false}
        style={
          Object {
            "height": 112,
          }
        }
      />
    </div>
  </div>
  <div
    data-css-tayj8l=""
  >
    Some sublabel
  </div>
</label>
`;
